rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/RholangMethodsCostsSpec.scala

```diff
@@ -0,0 +1,688 @@
+package coop.rchain.rholang.interpreter.accounting
+import java.nio.file.{Files, Path}
+
+import cats.Id
+import cats.effect.Sync
+import com.google.protobuf.ByteString
+import coop.rchain.models.Expr.ExprInstance._
+import coop.rchain.models._
+import coop.rchain.rholang.interpreter._
+import org.scalatest.{Assertion, BeforeAndAfterAll, WordSpec}
+import coop.rchain.models.rholang.implicits._
+import monix.eval.Task
+import org.scalatest.prop.PropertyChecks._
+
+import scala.concurrent.duration._
+import monix.execution.Scheduler.Implicits.global
+import org.scalactic.TripleEqualsSupport
+import coop.rchain.models.testImplicits._
+import coop.rchain.rholang.interpreter.Runtime.{RhoContext, RhoISpace}
+import coop.rchain.rholang.interpreter.accounting.Chargeable._
+import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
+import coop.rchain.rspace.history.Branch
+import coop.rchain.rspace.{Context, ISpace, RSpace}
+import org.scalatest.prop.TableFor1
+
+import scala.collection.immutable.BitSet
+
+class RholangMethodsCostsSpec extends WordSpec with TripleEqualsSupport with BeforeAndAfterAll {
+
+  "nth method" when {
+    "called on a list" should {
+      "charge constant cost regardless of collection size or value of n" in {
+        implicit val errorLog = new ErrorLog()
+        val table = Table(
+          ("list", "index"),
+          (List[Par](GInt(1)), 0),
+          (List[Par](GInt(1), GInt(2)), 1),
+          (Seq.fill[Par](10000)(GInt(1)).toList, 1000)
+        )
+        forAll(table) {
+          case (list, arg) =>
+            test(methodCall("nth", EList(list), List(GInt(arg))), NTH_METHOD_CALL_COST)
```

Comment:
```scala
(target, args) => EMethod(_method-name_, target, args)
```
won't be very useful here because input table to such test would be unnecessarily bloated. Particular methods require specific input, more than that - not every (target, args) pair is valid one for every method.